core/src/main/java/org/elasticsearch/index/analysis/AnalysisService.java

```diff
@@ -262,10 +261,13 @@ public AnalysisService(Index index, @IndexSettings Settings indexSettings, @Null
             }
         }
 
-        defaultAnalyzer = analyzers.get("default");
+        NamedAnalyzer defaultAnalyzer = analyzers.get("default");
         if (defaultAnalyzer == null) {
             throw new IllegalArgumentException("no default analyzer configured");
         }
+        if (analyzers.containsKey("default_index") && Version.indexCreated(indexSettings).onOrAfter(Version.V_2_0_0)) {
```

Comment:
Why 2.0+? Shouldn't this be (for master):
```
if (<3.0) {
  // log deprecation
} else {
  // error
}
```

Even for backporting, if a user has an old index, it should still show up as deprecated?